scrapy/spidermiddlewares/stickymeta.py

```diff
@@ -0,0 +1,31 @@
+from scrapy.exceptions import NotConfigured
+from scrapy.http import Request
+
+
+class StickyMetaParamsMiddleware(object):
+    """Forward a configurable list of meta keys through subsequent requests"""
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        keys_to_sticky = getattr(crawler.spider, 'sticky_meta_keys', [])
```

Comment:
I wonder if we shouldn’t use a setting instead. It would allow project-wide sticky meta keys, which may be useful to configure third-party Scrapy extension, and `Spider.custom_settings` it would make `Spider.sticky_meta_keys` unnecessary.

Reply:
I thought about this and I think you are right, I was thinking in my workflow where every spider is very unique, but the common use case is probably configuring a sticky param for every spider.
I didn't think about using `custom_settings` before, it can easily replace what I had in mind.
I made this change.